datajoint/expression.py

```diff
@@ -187,6 +217,13 @@ def restrict(self, restriction):
         string, or an AndList.
         """
         attributes = set()
+        if isinstance(restriction, Top):
+            self._top = dict(
```

Comment:
Why convert into a dict? Is `Top` not good enough?  You can use `dataclasses.asdict` too if wish to convert.

Reply:
You are correct. I did not think of this :sweat_smile: